src/rgw/rgw_dedup.h

```diff
@@ -0,0 +1,239 @@
+#pragma once
+#include "common/dout.h"
+#include "rgw_common.h"
+#include "rgw_dedup_utils.h"
+#include "rgw_dedup_table.h"
+#include "rgw_dedup_cluster.h"
+#include "rgw_realm_reloader.h"
+#include <string>
+#include <unordered_map>
+#include <variant>
+#include <iostream>
+#include <ostream>
+using namespace std;
+
+namespace rgw::dedup {
+  struct dedup_epoch_t;
+  struct control_t {
+    control_t() {
+      reset();
+    }
+    void reset();
+    inline bool local_urgent_req() const {
+      return (shutdown_req || local_pause_req);
+    }
+    inline bool should_stop() const {
+      return (shutdown_req || remote_abort_req);
+    }
+    inline bool should_pause() const {
+      return (local_pause_req || remote_pause_req);
+    }
+    void show_dedup_type(const DoutPrefixProvider* const dpp, const char* caller);
+
+    // TBD: pack state data members in a struct
+    // allow to start/pasue/resume/stop execution
+    int  dedup_type         = 0;
```

Comment:
what are the different types?
maybe use an enum instead?

Reply:
done